### PR TITLE
feat(train): more efficient gradient accumulation

### PR DESCRIPTION
**Pull request by borisdayma**
Previously we were using `optax.apply_every` which calls the optimizer with no gradient and could be an issue for some of them.
We tried `optax.MultiSteps` but it required too much memory at compilation so ended up building a custom loop.

### CODE REVIEW

**Comment by borisdayma** [comment]
Seems to work fine. I divide bs by 2 and accumulate it over 2 steps.
Interesting to note that the speed is barely affected!

image
